File: app/src/main/java/com/omar/retromp3recorder/app/repo/StateRepo.java
package com.omar.retromp3recorder.app.repo;

import com.omar.retromp3recorder.app.main.MainView;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Observable;
import io.reactivex.subjects.BehaviorSubject;

@Singleton
public class StateRepo {
    private final BehaviorSubject<MainView.State> stateSubject = BehaviorSubject
            .createDefault(MainView.State.Idle);

    @Inject
    public StateRepo(){}

    public void newValue(MainView.State state) {
        stateSubject.onNext(state);
    }

    public Observable<MainView.State> observe() {
        return stateSubject;
    }
}

File: app/src/main/java/com/omar/retromp3recorder/app/common/repo/StateRepo.kt
package com.omar.retromp3recorder.app.common.repo

import com.omar.retromp3recorder.app.main.MainView
import io.reactivex.Observable
import io.reactivex.subjects.BehaviorSubject
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class StateRepo @Inject constructor() {
    private val stateSubject = BehaviorSubject
        .createDefault(MainView.State.Idle)

    fun newValue(state: MainView.State) {
        stateSubject.onNext(state)
    }

    fun observe(): Observable<MainView.State> {
        return stateSubject
    }
}